### PR TITLE
feat(example): add example for stream api

### PR DESCRIPTION
**Pull request by kdada**
**What this PR does**
Add example for stream api.

**why we need it**
To show how we transfer big data.

**Special notes for your reviewer**:
@Jimexist

### CODE REVIEW

File: examples/stream/stream/create.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+package stream
+
+import (
+	"bytes"
+	"context"
+	"io"
+)
+
+func Stream(ctx context.Context) (io.Reader, error) {
```
Comment (ddysher) [inline examples/stream/stream/create.go]: The framework is responsible to read and stream the data, and the 'interface' we provide to framework user is `definition.Data`?

Comment (kdada) [reply examples/stream/stream/create.go]: Yes. Whatever the type of the returned value is, for users, it's `Data`.

File: examples/stream/api/v1/stream.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	register(app)
+}
+
+var app = definition.Descriptor{
+	Path: "/stream",
+	Definitions: []definition.Definition{
+		{
+			Method:   definition.Create,
+			Function: stream.Stream,
+			Produces: []string{"application/somedata"},
```
Comment (Jimexist) [inline examples/stream/api/v1/stream.go]: what does this mean?

Comment (kdada) [reply examples/stream/api/v1/stream.go]: I just define a content type to explain how to use producer to handle `io.Reader`.

**Comment by Jimexist** [comment]
merge to move forward

**Pull request merged**